### ISSUE TITLE
Add troubleshooting to documentation

### ISSUE THREAD

**Issue by kinow** [cylc/cylc-flow#3152]
Being the one with less experience on troubleshooting Cylc, it has always been challenging when a cylc workflow did not behave the way I expected it to.

Other tools include sections with tips for admins and developers to troubleshoot the tools. Maybe we could do the same? Here are some cases that could be documented (feel free to edit this or add more comments):

- [ ] troubleshoot problems building the project with `pip`, `virtualenv`, `conda`, etc. e.g. if you have a `build` directory from a previous `python setup.py bdist` it may interfere when using virtual environments

**Comment by hjoliver** [comment]
Good idea.  Finding developer time to do this is of course the issue at the moment though ... there's a potentially super large number of things that could/should be documented in this way.  

I guess we should start a Troubleshooting document (or User Guide section? or web page on the main site?) and just start populating it slowly though ... and eventually it will build up.

**Comment by hjoliver** [comment]
I did start an FAQ on the web site once, but ended up deleting it because we never got around to adding enough info to it.

**Comment by kinow** [comment]
>Good idea. Finding developer time to do this is of course the issue at the moment though ... there's a potentially super large number of things that could/should be documented in this way.

Yes, I set the milestone "some day" exactly because we are busy with other things. The idea here is to copy the issue to fix the flaky tests. While we do not have the bandwidth to fix that now, we have been documenting what needs to be done there. That should hopefully help whoever works on that later :crossed_fingers: .

>I guess we should start a Troubleshooting document (or User Guide section? or web page on the main site?) and just start populating it slowly though ... and eventually it will build up.

+1, GitHub Wiki, cylc-admin, user guide. Anywhere we can add quick notes and refine later should work IMO.
